### PR TITLE
fix R_Square shape issue in model.evaluate

### PR DESCRIPTION
**Pull request by JakeTheWise**
# Description

The `R_Square` metric was broken when calling `model.evaluate` or `model.fit` with `validation_data` set. This was due to a tensor shape mismatch.

Fixes # (issue)

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*  I added to `metrics/test/r_square_test.py`:
```python
def test_keras_fit():
    model = tf.keras.Sequential([tf.keras.layers.Dense(1)])
    model.compile(loss="mse", metrics=[RSquare(y_shape=(1,))])
    data = tf.data.Dataset.from_tensor_slices(
        (tf.random.normal(shape=(100, 1)), tf.random.normal(shape=(100, 1)))
    )
    data = data.batch(10)
    model.fit(x=data)
    model.fit(x=data, validation_data=data)
```

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [x] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

### CODE REVIEW

**Comment by bhack** [comment]
Thanks. See the first point on the checklist.

**Comment by marload** [comment]
Thank You!

Could you solve the test failure?

First of all, flake8 will be solved by removing the import of `weights_broadcast_ops`, which is an unused function.
Also, the `test_r2_sklearn_comparison` has an error.

**Comment by JakeTheWise** [comment]
@marload I believe I made all the necessary changes. All the tests are passing except one build... is this my fault? Please advise.

**Comment by marload** [comment]
The failed test seems to fail in all other PRs. I think you'd better write an Issue.
